### PR TITLE
[bug 1181360] Throttle the number of counted clicks

### PR DESCRIPTION
**Pull request by glogiotatidis**

### CODE REVIEW

**Comment by glogiotatidis** [comment]
The point is to limit the number of clicks that are being tracked to prevent DOSing our message queues and workers.

@pmclanahan  r?

File: affiliates/settings/base.py [hunk 1]
```diff
@@ -430,3 +430,7 @@ def facebook_app_url_lazy():
 
 # List Of Locales Where Firefox Friends Messaging Is Translated & Enabled
 FRIENDS_MSG_LOCALES = ['cs', 'de', 'en-US', 'fr', 'nl', 'pt-BR', 'sl', 'sr']
+
+# Limit to three counts per minute.
+CLICK_THROTTLE_TIMEOUT = 60  # One minute
+MAX_CLICK_COUNT = 3
```
Comment (pmclanahan) [inline affiliates/settings/base.py]: With entire offices or companies sharing a single external IP, only allowing 3 clicks per minute per IP seems a bit restrictive. Not sure how many the system can handle, but I'm sure 10 or even 30 per min would be okay.

**Comment by pmclanahan** [comment]
Looks good. My main reservation is lack of tests for the new logic. Since this is a small single item that you needed to rate limit I can see doing it yourself, but I do want to ask if you considered just using [django-ratelimit](https://django-ratelimit.readthedocs.org/en/latest/).

**Comment by glogiotatidis** [comment]
I did consider `django-throttle-request` which -as django-ratelimit does- limits the access to the view. In this case I just want to limit the number of clicks we count and not block the whole view. In other words even if the limit is hit the view will still return what was supposed to return but the click won't be counted.

Since this is not blocking the view I picked numbers that reflect the amount of  genuine clicks on our banners per IP and per minute. Judging from the current affiliates leaderboard I don't believe that anyone has a website or blog that can drive more than that.

**Comment by pmclanahan** [comment]
Sounds good. I know you didn't want to limit the view itself, but django-ratelimit has a helper function (`is_ratelimited`) that will do what you want. This looks fine though and the lib is surely overkill. Just FYI though:

https://django-ratelimit.readthedocs.org/en/latest/usage.html#helper-function

**Comment by glogiotatidis** [comment]
Indeed useful helper. I added some tests and this is ready to go.

**Comment by pmclanahan** [comment]
Looks great! Nice work!

**Pull request merged**